tests/phpunit/StorageWriterTest.php

```diff
@@ -160,21 +162,153 @@ public function testInvalidToken() : void
         $configFile = [
             'parameters' => [
-                '#token' => 'invalid',
+                '#token' => getenv('KBC_TEST_TOKEN'),
                 'url' => getenv('KBC_TEST_URL'),
+                'incremental' => true,
             ],
             'storage' => [
                 'input' => [
                     'tables' => [
                         [
                             'source' => 'in.c-main.some-source',
-                            'destination' => 'some-table-1',
+                            'destination' => 'some-table-5',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        $app->run();
+        self::assertTrue($this->client->tableExists(getenv('KBC_TEST_BUCKET') . '.some-table-5'));
+        $data = $this->client->getTableDataPreview(getenv('KBC_TEST_BUCKET') . '.some-table-5');
+        $data = explode("\n", $data);
+        sort($data);
+        self::assertEquals(
+            [
+                '',
+                '"1","Bar"',
+                '"1","foo"',
+                '"2","Kochba"',
+                '"3","Foo"',
+                '"4","foobar"',
+                '"id","name"',
+            ],
+            $data
+        );
+    }
+
+    public function testEmptyManifest(): void
+    {
+        $temp = new Temp('wr-storage');
+        $temp->initRunFolder();
+        $baseDir = $temp->getTmpFolder();
+        $fs = new Filesystem();
+        $fs->mkdir($baseDir . '/in/tables/');
+        $tableName = $baseDir . '/in/tables/some-table-6';
+        $fs->dumpFile($tableName, "\"id\",\"name\"\n\"1\",\"Bar\"\n\"2\",\"Kochba\"\n\"3\",\"Foo\"\n");
+        $manifest = [];
+        $fs->dumpFile($tableName . '.manifest', \GuzzleHttp\json_encode($manifest));
+        $configFile = [
+            'parameters' => [
+                '#token' => getenv('KBC_TEST_TOKEN'),
+                'url' => getenv('KBC_TEST_URL'),
+            ],
+            'storage' => [
+                'input' => [
+                    'tables' => [
+                        [
+                            'source' => 'in.c-main.some-source',
+                            'destination' => 'some-table-6',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        $app->run();
+        self::assertTrue($this->client->tableExists(getenv('KBC_TEST_BUCKET') . '.some-table-6'));
+        $table = $this->client->getTable(getenv('KBC_TEST_BUCKET') . '.some-table-6');
+        self::assertEquals([], $table['primaryKey']);
+    }
+
+    public function testAlreadyExistsWrongPk(): void
```

Comment:
Chybí mi test, co se stane, pokud najde duplicitní PK. Tohle má sice duplicitní PK pokud by byl PK "id", ale PK v tabulce je name a tam konflikt není.

Reply:
je tam - `b`, `c`

Reply:
Jo, sakra, už to vidim. Ztratil jsem se v backslashích :)

Reply:
Chápu to dobře tak, že teď nemám možnost writerem založit tabulku s jiným PK než je v manifestu stažené tabulky?

Reply:
Pak mi ještě chybí testy na to, co se stane, když cílová tabulka má jiné PK nebo jiné sloupce (nebo řekneme, že tohle má řešit Client?).

Reply:
nemas. Pokud cilova tabulka ma PK, tak se nic nezmeni

Reply:
tak bych ještě doplnil, že pokud se PK zdrojový a cílový liší, mělo by to zařvat a nepokračovat.

Reply:
hotovo, i kdyz nevim jestli to teda ted neni moc prisny

Reply:
je to takhle v transformacích i v docker runneru, může to způsobit divný chování.